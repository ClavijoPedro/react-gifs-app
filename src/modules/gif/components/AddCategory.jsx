import { useState } from "react"


export const AddCategory = ({onNewCategory}) => {
  
    const [inputValue, setInputValue] = useState('')
    

    const onImputChange = ({target}) => {
        
        setInputValue(target.value)

    }

    const onSubmit =(e) => {
       
        e.preventDefault()
        
        if(inputValue.trim().length <=1) return; 

        onNewCategory(inputValue.trim())
    
        setInputValue('')
    
    }


    return (
        <form onSubmit={onSubmit} aria-label="form">

            <input
                name="gifSearch"
                type="text"
                placeholder="Buscar Gifs"
                onChange={onImputChange}
                value={inputValue}
            />

        </form>
    )
}
