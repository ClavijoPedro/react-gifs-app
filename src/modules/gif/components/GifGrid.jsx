import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";


export const GifGrid = ({ category }) => {
    
    const {images, isLoading} = useFetchGifs(category)
    
    return (
        <>
          <h3 className="category-name">{category}</h3>
          
          {isLoading && <p className="spinner"></p>}

          <div className="card-grid">

            {images.map((img => (

                <GifItem key={img.id} {...img} />  

            )))}

          </div>

  
        </>
    );
};
