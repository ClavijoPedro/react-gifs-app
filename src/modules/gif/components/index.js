// ARCHIVO DE BARRIL PARA EXPORTAR TODO EN UN UNICO ARCHIVO Y NO TENER MIL IMPORTS EN LOS COMPONENTES

export * from './AddCategory'
export *from './GifGrid'
export * from './GifItem'