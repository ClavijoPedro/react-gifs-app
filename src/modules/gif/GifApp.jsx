import { useState } from "react";
import { AddCategory, GifGrid } from "./components";


export const GifApp = () => {
  const [categories, setCategories] = useState(['ironman'])
  
  const onAddCategory = (newCategory) => {

    if(categories.includes(newCategory)) return;

    setCategories([newCategory, ...categories])
  }
 
  return (
    <div className="app-container">

      <header>
        <img className="logo" src="logo.png" alt="logo" />
      </header>

      
      <main>

        <AddCategory onNewCategory={onAddCategory} />
        
      
        {categories.map((category, index ) => (

          <GifGrid category={category} key={index}/>

        ))}

      </main>


      <footer>
        <a href="https://github.com/ClavijoPedro" target="_blank">
          <img src="github.png" alt="github" style={{width:'40px'}} />
        </a>
      </footer>
    
    </div>
  );
};


