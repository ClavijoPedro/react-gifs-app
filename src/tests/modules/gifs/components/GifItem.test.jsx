import { render, screen } from "@testing-library/react"
import { GifItem } from "../../../../modules/gif/components"

describe('Pruebas en GifItem component', () => {

    const title = 'Ironman'
    const url = 'https://ironman.com/iron.jpg' 


    test('El snapshot deberia coincidir', () => {

        const {container} = render(<GifItem title={title} url={url} />)

        expect(container).toMatchSnapshot()
    })

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        
        render(<GifItem title={title} url={url} />)

        // pruebo que el src y alt del tag img sean los correctos. (getByRole trae tags y atributos de los tags)

        // expect(screen.getByRole('img').src).toBe(url) // forma larga
        // expect(screen.getByRole('img').alt).toBe(title) // forma larga

        const {src, alt} = screen.getByRole('img') // forma corta desestructuro los attributos sin repetir linea

        expect(src).toBe(url)
        expect(alt).toBe(title)
     })

     test('debe de mostrar el titulo en el componente', () => { 
        
        render(<GifItem title={title} url={url} />)

        // espero que imprima el title en algun tag del componente (no busca en los atributos alt o url)
        expect(screen.getByText(title)).toBeTruthy()
      
    })

})