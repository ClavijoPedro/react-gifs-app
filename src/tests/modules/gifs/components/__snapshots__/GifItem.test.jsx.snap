// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en GifItem component El snapshot deberia coincidir 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Ironman"
      src="https://ironman.com/iron.jpg"
    />
    <p>
      Ironman
    </p>
  </div>
</div>
`;
